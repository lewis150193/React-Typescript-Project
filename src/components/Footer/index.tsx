import React from 'react';


const styles = {
    backgroundColor: '#607D8B',
    boxShadow: '0 1px 1px 0 rgba(0, 0, 0, 0.12)',
    boxSizing: 'border-box',
    width: '100%',
    color: 'white',
    textAlign: 'center',
    font: 'normal 18px sans-serif',
    padding: '45px',
    marginTop: '80px'
};

export const Footer = () => (
// @ts-ignore
    <div style={styles}>
        <span>Footer</span>
    </div>

)