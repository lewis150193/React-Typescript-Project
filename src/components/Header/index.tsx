import React from 'react';

    const styles = {
        backgroundColor: '#607D8B',
        boxShadow: '0 1px 1px 0 rgba(0, 0, 0, 0.12)',
        boxSizing: 'border-box',
        width: '100%',
        color: 'white',
        textAlign: 'center',
        font: 'normal 18px sans-serif',
        padding: '45px',
        marginTop: '80px'
    }

export const Header = () => (
// @ts-ignore
    <div style={styles}>
        <span>Header</span>
    </div>
);