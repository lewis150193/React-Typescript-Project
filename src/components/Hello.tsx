import * as React from "react";

interface MyProps {
    name: string;
}

interface extendedProps extends  MyProps{
    age: number
}
export const Hello = (props: extendedProps) => <h1>Hello {props.name}my age is {props.age}</h1>;