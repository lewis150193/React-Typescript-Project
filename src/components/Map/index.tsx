import React, { Component } from 'react';
import GoogleMapReact from 'google-map-react';

const AnyReactComponent = ({ text }: any) => <div>{text}</div>;

    interface Props {
        center: {
            lat: number,
            lng: number
        }
        text: string
    }

class SimpleMap extends Component<Props> {
    constructor(props: any) {
        super(props);

    }

    render() {
        return (
            // Important! Always set the container height explicitly
            <div style={{ height: '50vh', width: '100%' }}>
                <GoogleMapReact
                    bootstrapURLKeys={{ key:'AIzaSyAc74MwP4jB-tL0MB_WG8ItOZXwfgyFLac' }}
                    defaultCenter={this.props.center}
                    defaultZoom={11}
                >
                    <AnyReactComponent
                        lat={this.props.center.lat}
                        lng={this.props.center.lng}
                        text={this.props.text}
                    />
                </GoogleMapReact>
            </div>
        );
    }
}

export default SimpleMap;