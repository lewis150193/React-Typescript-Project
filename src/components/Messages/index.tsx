import React from "react";
import { MessageBoxProps } from "./types";
import { Div } from "../../styled-components/div";

export const MessageBox = ({ date, title, value }: MessageBoxProps) => (
  <Div>
    <p>
      <strong>Title:</strong>
      {title}
    </p>
    <p>
      <strong>Message:</strong>
      {value}
    </p>
    <p>
      <strong>Id:</strong>
      {date}
    </p>
  </Div>
);
