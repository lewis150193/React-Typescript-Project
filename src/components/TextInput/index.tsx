
import React from 'react';
import {TextInputProps} from "./types";

export const TextBox = ({ onChange, name, value, className, placeHolder}: TextInputProps) => (
        <input className={className} type='input' onChange={onChange} name={name}  value={value} placeholder={placeHolder}/>
    );

