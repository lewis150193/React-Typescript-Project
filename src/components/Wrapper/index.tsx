import React, {ReactNode} from 'react';
import {Footer} from "../Footer";
import {Header} from "../Header";

type Props = {
    children: ReactNode
}
export const Wrapper = ({children}: Props) => (
    <>
        <Header/>
        <div>{children}</div>
        <Footer/>
    </>
)