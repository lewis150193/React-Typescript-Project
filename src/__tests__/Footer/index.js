import { shallow } from "enzyme/build";
import Enzyme from 'enzyme/build'
import Adapter from 'enzyme-adapter-react-16/build'
import React from "react";
import {Footer} from "../../components/Footer";

Enzyme.configure({ adapter: new Adapter()});

describe('Should snapshot footer',  () => {
    it('should render footer', () => {
        const wrapper = shallow(<Footer />);
        expect(wrapper).toMatchSnapshot();
    });
});