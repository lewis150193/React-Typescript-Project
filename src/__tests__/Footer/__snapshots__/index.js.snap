// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should snapshot footer should render footer 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span>
        Footer
      </span>,
      "style": Object {
        "backgroundColor": "#607D8B",
        "boxShadow": "0 1px 1px 0 rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "white",
        "font": "normal 18px sans-serif",
        "marginTop": "80px",
        "padding": "45px",
        "textAlign": "center",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Footer",
      },
      "ref": null,
      "rendered": "Footer",
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          Footer
        </span>,
        "style": Object {
          "backgroundColor": "#607D8B",
          "boxShadow": "0 1px 1px 0 rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "white",
          "font": "normal 18px sans-serif",
          "marginTop": "80px",
          "padding": "45px",
          "textAlign": "center",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Footer",
        },
        "ref": null,
        "rendered": "Footer",
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
